superset/assets/src/components/FilterableTable/FilterableTable.jsx

```diff
@@ -167,7 +179,95 @@ export default class FilterableTable extends PureComponent {
     );
   }
 
-  render() {
+  renderGridCellHeader({ columnIndex, key, style }) {
+    const label = this.props.orderedColumnKeys[columnIndex];
+    const className = this.props.expandedColumns.indexOf(label) > -1
+      ? 'header-style-disabled'
+      : 'header-style';
+    return (
+      <TooltipWrapper key={key} label="header" tooltip={label}>
+        <div
+          style={{ ...style, top: style.top - 4 }}
+          className={`${className} grid-cell grid-header-cell`}
+        >
+          {label}
+        </div>
+      </TooltipWrapper>
+    );
+  }
+
+  renderGridCell({ columnIndex, key, rowIndex, style }) {
+    const columnKey = this.props.orderedColumnKeys[columnIndex];
+    return (
+      <div
+        key={key}
+        style={{ ...style, top: style.top - 4 }}
+        className={`grid-cell ${this.rowClassName({ index: rowIndex })}`}
+      >
+        {this.list.get(rowIndex)[columnKey]}
+      </div>
+    );
+  }
+
+  renderGrid() {
+    const { orderedColumnKeys, overscanColumnCount, overscanRowCount, rowHeight } = this.props;
+
+    let { height } = this.props;
+    height -= GRID_EXTRA_HEIGHT;
+    let totalTableHeight = height;
+    if (this.container && this.totalTableWidth > this.container.clientWidth) {
+      // exclude the height of the horizontal scroll bar from the height of the table
+      // and the height of the table container if the content overflows
+      height -= SCROLL_BAR_HEIGHT;
+      totalTableHeight -= SCROLL_BAR_HEIGHT;
+    }
+
+    const getColumnWidth = ({ index }) => this.widthsForColumnsByKey[orderedColumnKeys[index]];
+
+    // fix height of filterable table
+    return (
+      <ScrollSync>
+        {({ onScroll, scrollTop }) => (
+          <div
+            style={{ height }}
+            className="filterable-table-container Table"
+            ref={(ref) => { this.container = ref; }}
+          >
+            <div className="LeftColumn">
+              <Grid
+                cellRenderer={this.renderGridCellHeader}
+                columnCount={orderedColumnKeys.length}
+                columnWidth={getColumnWidth}
+                height={rowHeight}
+                ref={(ref) => { this.container = ref; }}
```

Comment:
Let's pull this out into a const so we aren't creating a new function every time. Comment applies elsewhere.

Reply:
+1